cmd/tempo-vulture/main.go

```diff
@@ -205,6 +207,54 @@ func main() {
 	log.Fatal(http.ListenAndServe(prometheusListenAddress, nil))
 }
 
+func emitBatches(c *jaeger_grpc.Reporter, t *traceInfo) {
+	log := logger.With(
+		zap.String("org_id", tempoOrgID),
+		zap.String("write_trace_id", fmt.Sprintf("%016x%016x", t.traceIDHigh, t.traceIDLow)),
+		zap.Int64("seed", t.timestamp.Unix()),
+	)
+
+	log.Info("sending trace")
+
+	for i := int64(0); i < generateRandomInt(1, batchHighWaterMark, t.r); i++ {
+		ctx := user.InjectOrgID(context.Background(), tempoOrgID)
+		ctx, err := user.InjectIntoGRPCRequest(ctx)
+		if err != nil {
+			log.Error("error injecting org id", zap.Error(err))
+			metricErrorTotal.Inc()
+			continue
+		}
+		err = c.EmitBatch(ctx, makeThriftBatch(t.traceIDHigh, t.traceIDLow, t.r, t.timestamp))
+		if err != nil {
+			log.Error("error pushing batch to Tempo", zap.Error(err))
+			metricErrorTotal.Inc()
+			continue
+		}
+	}
+
+}
+
+func queueFutureBatches(client *jaeger_grpc.Reporter, info *traceInfo) {
+	log := logger.With(
+		zap.String("org_id", tempoOrgID),
+		zap.String("write_trace_id", fmt.Sprintf("%016x%016x", info.traceIDHigh, info.traceIDLow)),
+		zap.Int64("seed", info.timestamp.Unix()),
+	)
+
+	if maybe(info.r) {
```

Comment:
i'd like to look at a better way to do this. perhaps we could generate a "total batches" int from the random number generator and store it in traceinfo. then we could decrement that value each time we emitted a batch?

Reply:
Good call, I think that cleans up the readability a bit.  I've made the update.